src/main/java/seedu/ibook/model/product/Price.java

```diff
@@ -12,11 +12,7 @@
     public static final String MESSAGE_CONSTRAINTS =
             "Prices should only be of type double, and should not be negative";
 
-    /*
-     * The first character of the name must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
-     */
-    public static final String VALIDATION_REGEX = "\\$?(\\d+(?:.\\d{1,2})?)";
+    public static final String VALIDATION_REGEX = "\\$?(\\d+(?:\\.\\d{1,2})?)";
```

Comment:
Redundant capture group
```suggestion
    public static final String VALIDATION_REGEX = "\\$?\\d+(?:\\.\\d{1,2})?";
```